qbsum/summarizers.py

```diff
@@ -1,396 +1,76 @@
+        :param max_length: Maximum number of words that the summary cans include.
+        :type max_length: int
+        :param lda: lambda parameter used in computation of MMR score.
         :type lda: float
-        :param IDF: IDF value of words of the document cluster.
-        :type IDF: dict
-        :return: best sentence among the sentences in the document cluster.
-        :rtype: Sentence
+        :return: the candidate summary
+        :rtype: list(str)
         """
+        complete_set = set()
+        for document in document_set:
+            complete_set = complete_set.union(set(document))
 
-        selectedSentences = [bestSentence]
-        sum_len = bestSentence.wordCount
-        
+        selected = collections.OrderedDict()
+        summary_len = 0
+        while set(selected) != complete_set and summary_len < max_length:
+            remaining = complete_set - set(selected)
+            mmr_score = lambda x: lda*self.__similarity(x, query) - (1-lda)*max([self.__similarity(x, y) for y in set(selected)-{x}], default=0)
+            next_selected = max(remaining, key=mmr_score) #self.__argmax(remaining, mmr_score)
+            summary_len += self.__count_words(next_selected)
+            selected[next_selected] = len(selected)
 
-        MMRval={}
+        if summary_file:
+            with open(str(summary_file), 'w', encoding='utf-8-sig') as f:
+                f.write(" ".join(selected))
 
-        # add sentences until number of words exceeds summary length
-        while (sum_len < nbWords):	
-            MMRval={}		
+        return list(selected)
 
-            for sent in sentences:
-                MMRval[sent] = self.__MMRScore(sent, query, selectedSentences, lda, IDF)
 
-            maxxer = max(MMRval, key=MMRval.get)
-            selectedSentences.append(maxxer)
-            sentences.remove(maxxer)
-            sum_len += maxxer.wordCount
-
-
-        summary = []
-        for sentence in selectedSentences:
-            summary.append(sentence.originalWords)
-        return summary
-
-
-    def __MMRScore(self, Si, query, Sj, lambta, IDF):
+    def __similarity(self, sent_1, sent_2):
         """
-        Function to calculate the MMR score given a sentence, the query and the current best set of sentences.
-        :param Si: particular sentence for which the MMR score has to be calculated.
-        :type Si: Sentence
-        :param query: reference query.
-        :type query: Sentence
-        :param Sj: already selected best sentences.
-        :type Sj: list(Sentence)
-        :param lambta: lambda value of the MMR formula.
-        :type lambta: float
-        :param IDF: IDF value of words of the document cluster.
-        :type IDF: dict
-        :return: MMR score.
+        Compute similarity score between two sentences.
+        :param sent_1: the first sentence.
+        :type sent_1: str
+        :param sent_2: the second sentence.
+        :type sent_2: str
+        :return: similarity score
```

Comment:
dans quel rang? Avec quelle sémantique? (je vois que c'est la même sémantique que dans `spacy`, mais juste parce que j'ai suivi ton code. Donc, je propose, eg, 
```suggestion
        :return: similarity score, in [0,1], where 1 == perfect similarity.
```
)